lib/src/cli.rs

```diff
@@ -609,16 +609,16 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
         }
     } else {
         let fetched = crate::deploy::pull(repo, imgref, None, opts.quiet).await?;
-        let staged_digest = staged_image.as_ref().map(|s| s.image_digest.as_str());
-        let fetched_digest = fetched.manifest_digest.as_str();
+        let staged_digest = staged_image.and_then(|s| s.digest().ok());
+        let fetched_digest = &fetched.manifest_digest;
         tracing::debug!("staged: {staged_digest:?}");
         tracing::debug!("fetched: {fetched_digest}");
         let staged_unchanged = staged_digest
-            .map(|d| d == fetched_digest)
+            .map(|d| d == *fetched_digest)
```

Comment:
Minor style nit I think coercing to a reference like `&d == fetched_digest` is perhaps more idomatic?

Reply:
Yeah I suppose so.  That's more my historic C-like brain wanting to dereference two pointers to compare the values instead of comparing the actual pointers.  But rust already [does the dereference for you](https://doc.rust-lang.org/1.81.0/src/core/cmp.rs.html#1660) (I had to go look up how that actually works!).